src/jsonSchema.ts

```diff
@@ -0,0 +1,98 @@
+import { appendErrors, Resolver, transformToNestObject } from 'react-hook-form';
+//import * as Ajv from 'ajv';
+import Ajv, { ErrorObject, Options } from 'ajv';
+import type { JSONSchema4, JSONSchema6, JSONSchema7 } from 'json-schema';
+import {
+  FieldError,
+  FieldErrors,
+  ResolverError,
+  ResolverSuccess,
+} from 'react-hook-form/dist/types/form';
+
+export type JSONSchema = JSONSchema4 | JSONSchema6 | JSONSchema7;
+
+const parseErrorSchema = <TFieldValues extends Record<string, any>>(
+  validationError: Array<ErrorObject> | null | undefined,
+  validateAllFieldCriteria: boolean,
+): FieldErrors<TFieldValues> =>
+  Array.isArray(validationError)
+    ? validationError.reduce(
+        (
+          previous: FieldErrors<TFieldValues>,
+          { dataPath, keyword, message },
+        ) => {
+          const path = dataPath.replace(/\//g, '.').replace(/^\./, '');
+          return {
+            ...previous,
+            ...(path
+              ? previous[path] && validateAllFieldCriteria
+                ? {
+                    [path]: appendErrors(
+                      path,
+                      validateAllFieldCriteria,
+                      previous as Record<string, any>,
+                      keyword,
+                      message,
+                    ),
+                  }
+                : {
+                    [path]:
+                      previous[path] ||
+                      ({
+                        type: keyword,
+                        message,
+                        ...(validateAllFieldCriteria
+                          ? {
+                              types: { [keyword]: message || true },
+                            }
+                          : {}),
+                      } as FieldError),
+                  }
+              : {}),
+          };
+        },
+        {},
+      )
+    : {};
+
+export interface JsonSchemaOptions {
+  ajvOptions?: Omit<
+    Options,
+    'allErrors' | 'async' | 'validateSchema' | 'transpile'
```

Comment:
is `allErrors` means to return all the errors? that probably suitable for criticiaMode: 'all' at useForm.

Reply:
I believe you mean `criteriaMode`.

As per [AJV's documentation](https://github.com/ajv-validator/ajv#validation-and-reporting-options):

> - _allErrors_: check all rules collecting all errors. Default is to return after the first error.

You're right. That's exactly it. I didn't remember that option on `useForm`. Anyway, how do I even get that useForm option from the resolver? I thought about the `context` (second argument of the resolver) but it's undefined.

Reply:
I assumed `validateAllFieldCriteria` is set true when `criteriaMode==='all'` and false otherwise (i.e. `firstError`). Already committed the change.